geode-core/src/main/java/org/apache/geode/internal/cache/CacheServerLauncher.java

```diff
@@ -578,46 +578,98 @@ public void running() {
     }
   }
 
-  private static Integer serverPort;
+  /**
+   * GEODE-5256: Parameters containing startup options specified by the user.
+   * Shared from ServerLauncher and available to all classes through static accessors.
+   */
+  public static class Parameters {
```

Comment:
We actually want to delete CacheServerLauncher.java (hopefully sooner than later). We only use it from ServerLauncher because CacheServerLauncher sets some ThreadLocals that Geode internals then read. The plan was to rip these out, find a better way to pass that info down into the internals and remove the use of CacheServerLauncher from ServerLauncher so that we can delete it. Is there some place else that we could move Parameters? Maybe a new ServerLauncherParameters class in distributed.internal.

Another, unrelated, change that I'd eventually like to see is moving the Launcher classes out of org.apache.geode.distributed to some other package that makes more sense. Maybe org.apache.geode.management or even a new package. We could temporarily have Launcher classes in distributed that extend the newly moved classes for backwards compatibility until Geode 2.0.

Reply:
Hello @kirklund,

Thanks for taking the time to reviewing this.
I'll try to work on the changes during next week, I'm a bit overloaded with other tasks at the moment so it might take me a while.
Cheers.